### PR TITLE
docs: clear up synthetic key documentation

### PR DESCRIPTION
**Pull request by agavra**
### Description 

Updating the synthetic key documentations to avoid making perspective claims of what will be, as we have not yet decided the final semantics for join keys.

### Testing done 
N/A

### Reviewer checklist
- [ ] Ensure docs are updated if necessary. (eg. if a user visible feature is being added or changed).
- [ ] Ensure relevant issues are linked (description should include text like "Fixes #<issue number>")

### CODE REVIEW

File: docs/developer-guide/joins/synthetic-keys.md [hunk 1]
```diff
@@ -15,27 +15,22 @@ CREATE TABLE OUTPUT AS
...
-that such a join would result in a table with a compound primary key containing both `L.ID` and
-`R.ID`, and that's what will happen once ksqlDB supports
-[structured keys][1].
-
-Unfortunately, until ksqlDB supports structured keys, the join must result in a table with a single
-primary key column. Because the previous is a full-outer join, either `L.ID` or `R.ID` may be 
-missing (`NULL`), or both may have the same value. But, the data produced to {{ site.aktm }}
-always has the message key set to the non-null `ID` column, as shown in the following table:
+performing a full outer join of two source tables, joining on their ID columns. In a 
+full-outer join, however, either (1) `L.ID` or `R.ID` may be missing (`NULL`) or (2) both
```
Comment (JimGalasyn) [inline docs/developer-guide/joins/synthetic-keys.md:19]:
```suggestion
full-outer join, either `L.ID` or `R.ID` may be missing (`NULL`), or both 
```

File: docs/developer-guide/joins/synthetic-keys.md [hunk 2]
```diff
@@ -15,27 +15,22 @@ CREATE TABLE OUTPUT AS
...
-performing a full outer join of two source tables, joining on their ID columns. You might expect
-that such a join would result in a table with a compound primary key containing both `L.ID` and
-`R.ID`, and that's what will happen once ksqlDB supports
-[structured keys][1].
-
-Unfortunately, until ksqlDB supports structured keys, the join must result in a table with a single
-primary key column. Because the previous is a full-outer join, either `L.ID` or `R.ID` may be 
-missing (`NULL`), or both may have the same value. But, the data produced to {{ site.aktm }}
-always has the message key set to the non-null `ID` column, as shown in the following table:
+performing a full outer join of two source tables, joining on their ID columns. In a
```
Comment (JimGalasyn) [inline docs/developer-guide/joins/synthetic-keys.md:18]:
```suggestion
performing a full outer join of two source tables, joining on their ID columns. But in a 
```

**Review by JimGalasyn — APPROVED** [review]
LGTM, with one suggestion. Do we want this fix to go into earlier versions?

**Comment by agavra** [comment]
@JimGalasyn - yeah, this fix can go into earlier versions as well. I don't know how urgent it is since I think it's just important that when we have structured keys people don't also see the "when we have structured" keys part of the docs

**Pull request merged**